subgraphs/_reference_/src/prices/oracles/AaveOracle.ts

```diff
@@ -4,31 +4,24 @@ import { CustomPriceType } from "../common/types";
 import { Address, BigDecimal, BigInt } from "@graphprotocol/graph-ts";
 import { AaveOracleContract } from "../../../generated/UniswapV2Factory/AaveOracleContract";
 
-export function getAaveOracleContract(network: string): AaveOracleContract {
-  return AaveOracleContract.bind(
-    constants.AAVE_ORACLE_CONTRACT_ADDRESS.get(network)
-  );
+export function getAaveOracleContract(
+  contractAddress: Address
+): AaveOracleContract | null {
+  if (utils.isNullAddress(contractAddress)) return null;
+
+  return AaveOracleContract.bind(contractAddress);
 }
 
-export function getTokenPriceFromAaveOracle(
-  tokenAddr: Address,
-  network: string
-): CustomPriceType {
-  const aaveOracleContract = getAaveOracleContract(network);
+export function getTokenPriceUSDC(tokenAddr: Address): CustomPriceType {
+  const config = utils.getConfig();
 
-  if (
-    constants.AAVE_ORACLE_CONTRACT_ADDRESS.get(network).equals(
-      constants.ZERO_ADDRESS
-    )
-  ) {
+  if (!config || config.aaveOracleBlacklist().includes(tokenAddr))
     return new CustomPriceType();
-  }
-  
-  if (!aaveOracleContract) {
-    return new CustomPriceType();
-  }
 
-  let tokenPrice: BigDecimal = utils
+  const aaveOracleContract = getAaveOracleContract(config.aaveOracle());
+  if (!aaveOracleContract) return new CustomPriceType();
+
+  const tokenPrice: BigDecimal = utils
     .readValue<BigInt>(
       aaveOracleContract.try_getAssetPrice(tokenAddr),
```

Comment:
Have you checked this? On mainnet aave v2 `getAssetPrice()` returns a value in ETH

Reply:
can you explain more? Mainnet does not support aave oracle method.